6-AdvancedScenarios/4-hybrid-SPA/App/client/src/App.jsx

```diff
@@ -0,0 +1,75 @@
+import React,{ useEffect, useState  } from 'react';
+import { Route, Routes, useLocation } from 'react-router-dom';
+import { InteractionRequiredAuthError } from "@azure/msal-browser";
+import { MsalProvider, useMsal } from "@azure/msal-react";
+import { Profile }  from './pages/Profile';
+import { PageLayout } from './components/PageLayout';
+import { Hello } from './pages/Hello';
+import { callApiToGetSpaCode } from './fetch';
+
+import "./styles/App.css";
+
+
+ 
+const Pages = () => {
+  return (
+    
+      <Routes>
+        <Route path='/profile' element={<Profile />}/>
+        <Route path='/hello' element={<Hello /> }/>
+      </Routes>
+  )
+} 
+
+
+
+export const App = ({ instance }) => {
+  const search = useLocation().search;
+  const getCode = new URLSearchParams(search).get('getCode');
+  const { inProgress } = useMsal();
+  const [data, setdata] = useState(null);
+
+  useEffect(() => {
+
+    if(getCode){
+      callApiToGetSpaCode()
+        .then((data) => {
+        if(inProgress === "none"){
+          // SPA auth code
+          let code = data.code;
+          instance.acquireTokenByCode({code})
+            .then((res) => {
+              setdata(res)
+            }).catch((error) => {
+              if(error instanceof InteractionRequiredAuthError){
+                 if (inProgress === "none") {
+                   instance.acquireTokenPopup({code})
```

Comment:
Code cannot be passed to acquireTokenPopup. I would instead invoke `loginPopup`, and provide `login_hint` or `sid`, as shown here: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/msal-browser-samples/HybridSample/views/client-redirect.hbs#L59

Reply:
@jasonnutter removed acquireTokenPopup and used loginPopup with login_hint or sid